### DISCUSSION TITLE
ZPL Displays Label in Labelary, but not in Virtual ZPL Printer

### DISCUSSION THREAD

**Discussion by jasonhun** [porrey/Virtual-ZPL-Printer#11] [General]
My ZPL brings up a label in Labelary, but in the Virtual ZPL Printer I get "ERROR: You requested the 1st label, but the specified ZPL doesn't generate any labels".

This is version 2.02.  I am able to send a basic one-liner label and have it display though.

**Comment by jasonhun** [comment]
It looks like it doesn't work with graphics/images. These work in labelary viewer, but I can't get any to come up that contain these commands to define/use an image.
~DGR:ImageName.GRF
^XGR:ImageName.GRF

**Reply by porrey** [reply]
Can you send an example of your ZPL and image to reproduce the issue?

**Reply by jasonhun** [reply]
Sure thing!  Here's an example.  This shows up fine in the Labelary viewer, but I can't seem to get it to come up in the Virtual ZPL printer.  

![symbol-ce-copy](https://user-images.githubusercontent.com/24816782/197292647-703ec71d-4bad-47e5-80e0-68192a6adddc.png)
[zplTest.txt](https://github.com/porrey/Virtual-ZPL-Printer/files/9842688/zplTest.txt)

Sending to the printer via C# using Zebra.Sdk.Comm

`
//Instantiate connection for ZPL TCP port at given address
Connection thePrinterConn = new TcpConnection(theIpAddress, TcpConnection.DEFAULT_ZPL_TCP_PORT);  
try
{
    //Open the connection - physical connection is established here.
    thePrinterConn.Open();
    //Send the data to printer as a byte array.
    thePrinterConn.Write(Encoding.UTF8.GetBytes(zpl));
}
catch (ConnectionException e)
{
    //Handle communications error here.
    Console.WriteLine(e.ToString());
}
finally
{
    //Close the connection to release resources.  
    thePrinterConn.Close();
}
`

**Reply by porrey** [reply]
It does work if you drop your **zplTest.txt** file in **Documents\Virtual ZPL Printer\Templates** and change the extension to **.zpl** and then use the test feature in the application. The test feature reads the ZPL from the file and sends it to the virtual printer over a TCP connection. This indicates the issue you are having could be in the exchange of data. I'll have to dig in further.

Do you get an error, blank label or something else?

**Reply by jasonhun** [reply]
I'll give that a try, thanks!

Yes, it gives this  message:

![image](https://user-images.githubusercontent.com/24816782/197296241-f36aad10-0f7b-41b8-8f8f-f3de544036bf.png)

**Reply by porrey** [reply]
That error comes back from Labelary. I can reproduce it if I remove the image but the leave the reference to it in (using the online viewer):

<img width="1157" alt="Screenshot 2022-10-21 at 5 25 30 PM" src="https://user-images.githubusercontent.com/12475115/197297341-6427374a-4979-456f-90fb-c4826588d29e.png">

**Comment by mbaggnx** [comment]
Hello there, happy new year in advance porrey! hope you had great start at 2k23

Askinfg about this, is there any way to fix on 2.2.0?

Thanks kind regards!

**Reply by porrey** [reply]
I am not clear on what the issue is.

**Reply by mbaggnx** [reply]
> I am not clear on what the issue is.

oh sorry jaja

just asking if this issue was fixed or if it is not (since i am sending images on the etiquete) and it keeps showing me this error.

![image](https://user-images.githubusercontent.com/119416732/210718093-5245d60c-68ef-45c1-b12f-e9c2f7fefb66.png)

**Comment by porrey** [comment]
I used the ZPL with an image (shown below) in the latest version and it worked fine.

```
^XA
^LH0,0
^CI28

~DGR:symbol-c.GRF,2300,20,
Q0G1IFGEg0JFP0G1JFGEY0KFP0KFGEX0G7KFO0LFGEW0G7LFN0G3LFGEV0G1MFN0MFGEV0G7MFM0G3MFGEU0G1NFM0NFGEU0G7NFL0G3NFGET0G1OFL0G7NFGET0G3OFK0G1OFGET0PFK0G3OFGES0G1PFK0G7OFGES0G3PFK0PFGES0G7PFJ0G1PFGER0G1QFJ0G7PFGER0G3QFJ0QFGER0G7QFI0G1OFG8T0OFGC,I0G1NFU0G1NFG8,I0G3MFV0G1MFG8,I0G7LFG8V0G3LFGC,I0MFW0G7LF,H0G1LFGCW0LFGC,H0G3LFW0G1LFG8,H0G3KFGCW0G1KFGE,H0G7KFG8W0G3KFGC,H0LFX0G7KFG8,H0KFGEX0G7KF,G0G1KFG8X0KFGE,G0G1KFX0G1KFGC,G0G3JFGEX0G1KFG8,G0G3JFGEX0G1KF,G0G7JFGCX0G3JFGE,G0G7JFG8X0G3JFGC,G0KFY0G7JFG8,:G1JFGEY0KF,G1JFGCY0JFGE,:G3JFG8X0G1JFGC,:G3JFY0G1JFG8,G7JFY0G3JFG8,:G7IFGEY0G3JF,:JFGEY0G7JF,JFGCY0G7IFGE,JFGCY0G7JF,JFGCY0G7RF,JFGCY0G7RFG8,JFG8Y0G7RF,::::::::::::JFGCY0G7RF,:JFGCY0G7JF,:JFGEY0G3JF,G7IFGEY0G3JF,:G7JFY0G3JFG8,G3JFY0G1JFG8,:G3JFG8X0G1JFGC,G1JFGCX0G1JFGE,G1JFGCY0JFGE,G1JFGEY0KF,:G0KFY0G7JFG8,G0KFG8X0G7JFGC,G0G7JFG8X0G3JFGC,G0G7JFGCX0G3JFGE,G0G3JFGEX0G1KF,G0G3KFX0G1KFG8,G0G1KFG8X0KFGC,G0G1KFGCX0KFGE,H0KFGEX0G7KF,H0G7KFX0G3KFG8,H0G7KFG8W0G3KFGC,H0G3KFGEW0G1LF,H0G1LFG8W0LFGC,H0G1LFGCW0LFGE,I0MFW0G7LFG8,I0G7LFGEV0G3MF,I0G3MFG8U0G1MFGC,I0G1NFG8U0NFGE,J0QFGCR0G7QFJ0G7PFGER0G3QFJ0G3PFGER0G1QFJ0G1PFGES0QFK0PFGES0G7PFK0G7OFGES0G3PFK0G3OFGET0PFL0OFGET0G7OFL0G7NFGET0G1OFL0G1NFGEU0OFM0G7MFGEU0G3NFM0G1MFGEV0NFN0MFGEV0G7MFN0G1LFGEW0MFO0G7KFGEW0G3LFP0G7JFGEX0G7KFQ0G7IFGEY0G3JFQ0G1IFGEg0JF

^FO50,50
^XGR:symbol-c.GRF,1,1^FS
^XZ
```

<img width="485" alt="Screenshot 2023-01-05 at 6 48 02 AM" src="https://user-images.githubusercontent.com/12475115/210783735-87aa99ba-5e84-4300-bea8-221f8401819c.png">

**Reply by mbaggnx** [reply]
Hi! tested your txt and works, but mine does not, ill make some tests based on your txt writing


Also, I do not have that zplTest, do you have any folder where we put .txt test to send? If yes, I am not able to see where after instalation

![image](https://user-images.githubusercontent.com/119416732/210803236-e5d0d89e-eea6-4a65-b8ef-5aea9dc0eca3.png)


Also I see a error once you close the Send Label Test it gets "bugged" if I am not wrong
![image](https://user-images.githubusercontent.com/119416732/210803558-0a0dd4d6-e471-4c51-857b-82acfb6202f4.png)

**Reply by porrey** [reply]
Documents\Virtual ZPL Printer\Templates. Use .zpl

**Comment by mbaggnx** [comment]
Ok! I just found the diference @porrey 

The diference is that your image is drawn diferent from mine even at the reading, wich it reads ok the logo, but gives the other error I think:

Header of picture reading:
^FT21,1199^GFA,5472,5472,12, 

Picture (has been shorted the picture):

000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
0000000000000FE000000000
0000000000003FE000000000
0000000000007FE010000000
000000000001FFE078000000
000000000003FFE0FC000000
000000000007FFE0FC000000
000000000007F1E07C000000
00000000000FE1E07C000000
00000000000FC0E03E000000
00000000000F80E03E000000
00000000000F81E03E000000
00000000001F80E03E000000
00000000001F01E03E000000
00000000001F01E03E000000
00000000001F80E03E000000
00000000000F81E07E000000
00000000000FC1E07C000000
00000000000FC1E0FC000000
00000000000FE0E1FC000000
000000000007F9E7F8000000
000000000007FFFFF8000000
000000000001FFFFF0000000
000000000001FFFFE0000000
000000000000FFFFC0000000
0000000000003FFF00000000
0000000000000FFC00000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
0000000000001FFFFC000000
0000000000007FFFFC000000
000000000001FFFFFC000000
000000000003FFFFFC000000
000000000007FFFFFC000000
00000000000FFFFFFC000000
00000000000FFE0000000000
00000000000FF00000000000
00000000000FC00000000000
00000000001FC00000000000
00000000001F800000000000
00000000001F800000000000
00000000001F800000000000
00000000000F800000000000
00000000000F800000000000
00000000000FC00000000000
00000000000FC00000000000
000000000003E00000000000
000000000003F00000000000
000000000001FFFFFC000000
000000000000FFFFFC000000
000000000001FFFFFC000000
000000000003FFFFFC000000
000000000007FFFFFC000000
00000000000FFFFFFC000000
00000000000FFFFFFC000000
00000000000FE00000000000
00000000000FC00000000000
00000000001FC00000000000
00000000001F800000000000
00000000001F800000000000
00000000000F800000000000
00000000000F800000000000
00000000000F800000000000
00000000000FC00000000000
000000000007C00000000000
000000000003E00000000000
000000000001F00000000000
00000000000FFFFFFC000000
00000000000FFFFFFC000000
00000000000FFFFFFC000000
00000000000FFFFFFC000000
00000000000FFFFFFC000000
00000000000FFFFFFC000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
0000000000000FFE00000000
0000000000003FFF80000000
0000000000007FFFC0000000
000000000001FFFFE0000000
000000000001FFFFF0000000
000000000003FFFFF8000000
000000000007FC07F8000000
00000000000FF003FC000000
00000000000FE000FC000000
00000000000FC000FC000000
00000000000FC0007E000000
00000000000FC0007E000000
00000000001F80007E000000
00000000001F80007E000000
00000000001F80007E000000
00000000001F80007E000000
00000000000FC0007E000000
00000000000FC0007E000000
00000000000FC000FC000000
00000000000FE000FC000000
00000000000FF001FC000000
000000000007FE0FF8000000
000000000003FFFFF8000000
000000000003FFFFF0000000
000000000001FFFFE0000000
000000000000FFFFC0000000
0000000000003FFF80000000
0000000000000FFC00000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
000000000000000000000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
000000000001FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
000000000000FC0000000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
00000000FFFFFFFFFC000000
000000000000000000000000
000000000000000000000000
